src/HasSlug.php

```diff
@@ -149,8 +149,14 @@ protected function makeSlugUnique(string $slug): string
      */
     protected function otherRecordExistsWithSlug(string $slug): bool
     {
+        $key = $this->getKey() ?? '0';
+
+        if (! $this->incrementing) {
+            $key = $this->getKey();
+        }
+
         return (bool) static::where($this->slugOptions->slugField, $slug)
-            ->where($this->getKeyName(), '!=', $this->getKey() ?? '0')
+            ->where($this->getKeyName(), '!=', $key)
```

Comment:
I changed it like this because I tested it in the specific situation and found that only when the incrementing value was false and using a PostgreSQL database it give the error from #63